app/gui/src/controller/ide/desktop.rs

```diff
@@ -164,3 +161,27 @@ impl ManagingProjectAPI for Handle {
         .boxed_local()
     }
 }
+
+/// Select a new name for the project in a form of <suggested_name>_N, where N is a unique sequence
+/// number.
+fn choose_new_project_name(
+    mut existing_names: impl Iterator<Item = String>,
+    suggested_name: &str,
+) -> String {
+    let with_suffix = (1..).map(|i| format!("{}_{}", suggested_name, i));
+    let mut candidates = std::iter::once(suggested_name.to_owned()).chain(with_suffix);
+    // The iterator have no end, so we can safely unwrap.
+    let name = candidates.find(|c| !existing_names.contains(c)).unwrap();
+    capitalize_first(&name)
+}
+
+/// Capitalize the first letter of the string.
+///
+/// CC-BY-SA 4.0 https://stackoverflow.com/a/38406885
```

Comment:
IANAL, but I don't think this counts as a proper attribution.
On the other hand I don't think that such trivial snippet should be even treated as a work in sense of copyright law.

Reply:
It's an extremely complicated legal stuff. I think this is considered a proper attribution, and this code sample will probably fall under "fair use" conditions. However, I changed it significantly while refactoring the code, and removed attribution, therefore. It takes too much effort to copy code from SO :(